ddtrace/appsec/_iast/_loader.py

```diff
@@ -32,6 +32,14 @@ def _exec_iast_patched_module(module_watchdog, module):
 
     if compiled_code:
         # Patched source is executed instead of original module
-        exec(compiled_code, module.__dict__)  # nosec B102
-    elif module_watchdog.loader is not None:
+        try:
+            exec(compiled_code, module.__dict__)  # nosec B102
+            return
+        except ImportError:
+            log.debug("Unexpected exception while executing patched code", exc_info=True)
+
+    # Fallback
+    try:
```

Comment:
Why are we removing the line `elif module_watchdog.loader is not None:`? Could you point out some cases where this was failing and the exception is necessary?

Reply:
Yes. This would totally crash while trying to exec grpc auto-generated modules. El `elif` is not necessary since it's at the end of the function and is a fallback.

Reply:
Done.